### ISSUE TITLE
Consider adding $focused to fieldState

### ISSUE THREAD

**Issue by scottadamsmith** [fergaldoyle/vue-form#91]
It would be nice to be able to have the option to only display the messages if the field is in focus. In our use case, we put a red border on the field permanently once it is invalid, but only show the message when they are focused on the field. It could look something like this:

`<field-messages name="firstName" show="($submitted || $touched) && $focused"></field-messages>`

I looked at the code briefly and it seems as though this might be possible using the _setFocused method here:

https://github.com/fergaldoyle/vue-form/blob/0bfcdbae582ef5dd05b1ae55f1e8b4fe411a9de1/src/components/validate.js#L145-L147

It would also require adding an additional state that represents whether a field is currently focused, rather than that it has been focused at some point.

I would be interested in contributing this if it seems worthwhile. Also, if I am missing an obvious roadblock here, please let me know.

**Comment by fergaldoyle** [comment]
Feel free to make this contribution. I wonder would  `$focus` or `$infocus` be better than `$focused` since is a current state rather than a past tense one.

**Comment by scottadamsmith** [comment]
Fair point. I figured there might be some back and forth on that state name. Given those options, I think I'd lean $infocus. What are your thoughts on $hasfocus. I got this idea from https://developer.mozilla.org/en-US/docs/Web/API/Document/hasFocus

**Comment by fergaldoyle** [comment]
$focus sounds good to me

**Comment by scottadamsmith** [comment]
@fergaldoyle I have a fork with the changes just about ready to submit. I did send out my changes to my team over here to review beforehand and I got some feedback around the naming. Here is the commit if you'd like to look at it for reference: 

https://github.com/ClickBoarding/vue-form/commit/db981c50ec3827815577c07a99c638e2979c0a28

The feedback that was given was around my use of the "hasFocus variable internally and how it differed from all the internal state variables that use the "is" suffix. The suggestion was to switch to isFocused for the internal variable and that got me thinking about the exposed state name again. 

You mentioned the concern of it being a current state as opposed to a past tense one, but I thought about the the existing states and it seems like they are current state, but the other states are less temporary than focus is. For example, I can see where someone would interpret "touched" as "was previously touched", but I think it could also be interpreted as a current state when you think of "touched" as opposed to "untouched". 

tl;dr: I am going to rename my internal variable from hasFocus to isFocused, but if you can get behind it, I'd like to also change the state name from $focus to my original proposal of $focused. Let me know what  your thoughts are and also if you have any other feedback before I send the PR.
